File: java/src/com/jambit/practicalresult/Recover.java
package com.jambit.practicalresult;

import io.vavr.collection.List;
import io.vavr.control.Either;

public class Recover {

  public static void main(String[] args) {
    final Either<String, List<Integer>> result = ListService.get(-1);

    System.out.println(result);

    final Either<String, List<Integer>> recovered = result.isRight()
                                                    ? result
                                                    : Either.right(List.empty());

    System.out.println(recovered);
  }
}

File: swift/Recover.swift

func recover() {
    printName()

    let result = listService(-1)

    print(result)

    let recovered = result.recover(with: [])

    print(recovered)
}

extension Result {

    func recover(with defaultValue: Success) -> Result<Success, Failure> {
        switch self {
        case .success: return self
        case .failure: return .success(defaultValue)
        }
    }
}
